### DISCUSSION TITLE
How to Assign a Custom Class in the new fast dreambooth method?

### DISCUSSION THREAD

**Discussion by MasterDenis** [TheLastBen/fast-stable-diffusion#456] [Q&A]
Basically the title, how is it possible to assign a custom class if we wish to in case we train something more specific?
and if it is not possible can we get it as an optional feature?
Thanks.

**Comment by TheLastBen** [comment]
the class is determined automatically, you don't need to specify it

**Reply by MasterDenis** [reply]
So if I wish to train in the model a more specific class how would that work? 
if it is not a face or if it is a face but without the normal human features, or even if it is a younger individual than the average man/woman archetype.

**Reply by TheLastBen** [reply]
try training on it with 3000 steps, try calling it with the instance name in A1111 and see if you get close results, if not, we'll find another solution

**Reply by MasterDenis** [reply]
to give you a perspective, I trained with a custom model (which by itself does not deviate too much from the base sd model by the official team), the next step was the recommended 4k steps for 20 images and everything else at default (literarly) and the model got overtaken by the face of the subject, anything would just print a similar image with just different colors.
Now I did not use the contain faces and I ran for 4k steps.

My perfect (so far) or most perfect result by using the base model would be in Shiv's Colab the default settings but with 100 REG images and 2250 steps (for 20 images), the results are very very close to the perfect likeness with average images (taken from a good phone), so far with the Shiv's Colab it is about images*112.5 for about 10-20 Images. (Which is your older options I think)
Anyway, Those were my test findings so far.
If I were to compare, the new fast method is a big more aggressive to let the concept overtake the model in some sense because given the recommended settings (even at 3000 steps for 20 images) the character/person will be looking very ugly in the sense that if they are a young woman, their natural face lines will make it look like an older woman at higher CFG Scales (which sometimes I need to set for prompting purposes).
The reason for 20 input images is because the person has a bit more complex face structure and I wish to add a bit more detail to the model (different backgrounds, same person, slight expression changes, slight angle changes etc)

I find it a good practice to return the old method (Shiv Colab?) with a few tweaks to make it simpler to use while also providing options, the ability to use any .ckpt file as a base is good already in your colab so the above would be a big plus.

Also the ability to assign classes and tokens of different concepts for parallel training of subjects would be useful, i.e. man and woman, or woman and girl (in case of a mother and daughter or something).
Thanks :)

**Reply by TheLastBen** [reply]
did you try finding the best settings using a celebrity's pictures as I recommended ? using your own pictures could mislead you settings wise.
here some examples with the standard settings, 4k-5k steps with 35% textenc and 2 subjects at once, I'm sure it's all about the instance pictures, not the colab.

![download (10)](https://user-images.githubusercontent.com/110583491/201371944-70b7cb89-f535-4cd5-bb30-bcf2f7c915be.png)
![download (d1)](https://user-images.githubusercontent.com/110583491/201371955-dadc89b7-aa55-4414-b93d-7ac530d37a14.png)
![download (9)](https://user-images.githubusercontent.com/110583491/201371858-1da83667-bb88-4127-8fff-e92e17601cfb.png)

![download (1)](https://user-images.githubusercontent.com/110583491/201371838-323a2b63-faa1-4f90-8b83-c55527ad3a99.png)

**Comment by TheLastBen** [comment]
and here some examples training on a old phoography style : 

![download (24)](https://user-images.githubusercontent.com/110583491/201372749-473e9805-d861-4c95-aa00-c15aa65037d6.png)
![tmp_4zhz_6f](https://user-images.githubusercontent.com/110583491/201372500-3dcb1a57-962f-419a-88bc-f279bc150ca2.png)
![tmp753me50h](https://user-images.githubusercontent.com/110583491/201372545-c34de580-488d-4f4e-b4de-b5a920707725.png)
![tmpjwy1iuf_](https://user-images.githubusercontent.com/110583491/201372571-45606188-edc4-4d4d-95da-fe80d7b7c5ff.png)
![tmprm_tg9ac](https://user-images.githubusercontent.com/110583491/201372611-692c6ed8-edf8-4c85-a655-724bfea1d0bb.png)
![download (1)](https://user-images.githubusercontent.com/110583491/201372694-e4bea2cc-4c5e-4dae-abb9-46727e0d2dea.png)
![download (8)](https://user-images.githubusercontent.com/110583491/201372719-dfe774f2-4032-4055-b55c-d2e05b46ab7c.png)

**Comment by TheLastBen** [comment]
More : 

![download (17)](https://user-images.githubusercontent.com/110583491/201373136-0523c75b-b1c2-401c-83cf-2cfb2568f57b.png)
![download](https://user-images.githubusercontent.com/110583491/201373164-9d241586-6a40-4a99-9ce4-010ca92f531b.png)
![download (2)](https://user-images.githubusercontent.com/110583491/201372903-37f6fc6e-18b0-4bc4-8cdc-cec91d03ceee.png)
![download (3)](https://user-images.githubusercontent.com/110583491/201372933-43c71638-7e3a-4407-b418-5328b16aaf8e.png)
![download (4)](https://user-images.githubusercontent.com/110583491/201372962-72b43945-03f3-4047-ac34-704d970a1b8a.png)
![download (8)](https://user-images.githubusercontent.com/110583491/201372989-8c58f897-692a-41f0-a9a6-7e20e901d0dc.png)
![download (9)](https://user-images.githubusercontent.com/110583491/201373032-ac9f1930-ae6a-46ac-85fc-b47aa8757226.png)
![download (14)](https://user-images.githubusercontent.com/110583491/201373050-14ccf974-b641-4437-9040-97a7bac2a475.png)
![download (16)](https://user-images.githubusercontent.com/110583491/201373092-b191f41c-74e8-4972-8155-6504f8832956.png)
